src/main/java/TypewiseAlert/TypewiseAlert.java

```diff
@@ -1,87 +1,41 @@
-      if(value < lowerLimit) {
-        return BreachType.TOO_LOW;
-      }
-      if(value > upperLimit) {
-        return BreachType.TOO_HIGH;
-      }
-      return BreachType.NORMAL;
-    }
-    public enum CoolingType {
-      PASSIVE_COOLING,
-      HI_ACTIVE_COOLING,
-      MED_ACTIVE_COOLING
-    };
-    public static BreachType classifyTemperatureBreach(
-        CoolingType coolingType, double temperatureInC) {
-      int lowerLimit = 0;
-      int upperLimit = 0;
-      switch(coolingType) {
-        case PASSIVE_COOLING:
-          lowerLimit = 0;
-          upperLimit = 35;
-          break;
-        case HI_ACTIVE_COOLING:
-          lowerLimit = 0;
-          upperLimit = 45;
-          break;
-        case MED_ACTIVE_COOLING:
-          lowerLimit = 0;
-          upperLimit = 40;
-          break;
-      }
-      return inferBreach(temperatureInC, lowerLimit, upperLimit);
-    }
-    public enum AlertTarget{
-      TO_CONTROLLER,
-      TO_EMAIL
-    };
-    public class BatteryCharacter {
-      public CoolingType coolingType;
-      public String brand;
-    }
-    public static void checkAndAlert(
-        AlertTarget alertTarget, BatteryCharacter batteryChar, double temperatureInC) {
+public class TypewiseAlert {
+public static BreachTypes inferBreach(double value, double lowerLimit, double upperLimit) {
+		if (value < lowerLimit) {
+			return BreachTypes.TOO_LOW;
+		}
+		if (value > upperLimit) {
+			return BreachTypes.TOO_HIGH;
+		}
+		return BreachTypes.NORMAL;
+	}
 
-      BreachType breachType = classifyTemperatureBreach(
-        batteryChar.coolingType, temperatureInC
-      );
+	public static boolean checkAndAlert(AlertTargets alertTargets, CoolingTypes coolingType, double temperature) {
+		BreachTypes breachTypes = inferBreach(temperature, coolingType.getLowerLimit(), coolingType.getUpperLimit());
+		return alertForBreach(alertTargets, breachTypes);
+	}
 
-      switch(alertTarget) {
-        case TO_CONTROLLER:
-          sendToController(breachType);
-          break;
-        case TO_EMAIL:
-          sendToEmail(breachType);
-          break;
-      }
-    }
-    public static void sendToController(BreachType breachType) {
-      int header = 0xfeed;
-      System.out.printf("%i : %i\n", header, breachType);
-    }
-    public static void sendToEmail(BreachType breachType) {
-      String recepient = "a.b@c.com";
-      switch(breachType) {
-        case TOO_LOW:
-          System.out.printf("To: %s\n", recepient);
-          System.out.println("Hi, the temperature is too low\n");
-          break;
-        case TOO_HIGH:
-          System.out.printf("To: %s\n", recepient);
-          System.out.println("Hi, the temperature is too high\n");
-          break;
-        case NORMAL:
-          break;
-      }
-    }
+	private static boolean alertForBreach(AlertTargets alertTarget, BreachTypes breachTypes) {
+		switch (alertTarget) {
+		case TO_CONTROLLER:
```

Comment:
Production code needs to decide which alert target needs to be called. This switch case can be eliminated by the use of interface for alert targets. The use of interface hence abstracts the alert targets from this production code, and in future if any new targets are added, there will be no code change required...